dot/rpc/modules/author_test.go

```diff
@@ -228,23 +274,27 @@ func TestAuthorModule_SubmitExtrinsic(t *testing.T) {
 			},
 			args: args{
 				req: &Extrinsic{fmt.Sprintf("0x%x", testExt)},
-				res: new(ExtrinsicHashResponse),
 			},
 			wantRes: ExtrinsicHashResponse(types.Extrinsic(testExt).Hash().String()),
 		},
 	}
 	for _, tt := range tests {
+		res := ExtrinsicHashResponse("")
+		tt.args.res = &res
 		t.Run(tt.name, func(t *testing.T) {
 			am := &AuthorModule{
 				logger:     tt.fields.logger,
 				coreAPI:    tt.fields.coreAPI,
 				txStateAPI: tt.fields.txStateAPI,
 			}
-			if err := am.SubmitExtrinsic(tt.args.r, tt.args.req, tt.args.res); (err != nil) != tt.wantErr {
-				t.Errorf("AuthorModule.SubmitExtrinsic() error = %v, wantErr %v", err, tt.wantErr)
-			}
-			if diff := cmp.Diff(tt.wantRes, *tt.args.res); diff != "" {
-				t.Errorf("unexpected response: %s", diff)
+			err := am.SubmitExtrinsic(tt.args.r, tt.args.req, tt.args.res)
+			if tt.wantErr {
+				if assert.NotNil(t, err) {
+					assert.EqualError(t, err, tt.err.Error())
+				}
```

Comment:
```suggestion
				require.Error(t, err) {
				assert.EqualError(t, err, tt.err.Error())
```

You can use `require` to make the test instantly fail and not continue instead. That usually reduces indentation and if/else clauses.